docs/data/data-grid/components/columns-panel/columns-panel.md

```diff
@@ -0,0 +1,41 @@
+# Data Grid - Columns Panel component 🚧
+
+<p class="description">Customize the columns panel UI.</p>
+
+:::warning
+This component is incomplete.
+
+Currently, the only feature available for the Columns Panel component is the Trigger. In the future, this component will allow you to extend the data grid's columns panel.
+
+In the meantime, see the following:
+
+- [Custom subcomponents—Columns panel](/x/react-data-grid/components/#columns-panel)
+  :::
```

Comment:
Keen to hear what others think about this note. I can foresee this composition pattern being helpful for the columns panel and filter panel when those eventually get a refresh. This note is in preparation for that.